viewer/db.js

```diff
@@ -1730,61 +1750,118 @@ Db.getIndices = async (startTime, stopTime, bounding, rotateIndex) => {
+        // the user-specified queryExtraIndices are less under our control, so we
+        //   are going to take some regex-based best guesses to figure out if it's hourly, daily, etc.
 
-      if (index[2] === 'w') {
-        len = 7 * 24 * 60 * 60;
-        month = 1;
-        day = (+index[3] * 10 + (+index[4])) * 7;
-      } else if (index[2] === 'm') {
-        month = (+index[3]) * 10 + (+index[4]);
-        day = 1;
-        len = 31 * 24 * 60 * 60;
-      } else if (index.length === 6) {
-        month = (+index[2]) * 10 + (+index[3]);
-        day = (+index[4]) * 10 + (+index[5]);
-        len = 24 * 60 * 60;
-      } else {
-        month = (+index[2]) * 10 + (+index[3]);
-        day = (+index[4]) * 10 + (+index[5]);
-        hour = (+index[7]) * 10 + (+index[8]);
-        len = hlength;
-      }
+        // daily 240311                         v year      v month        v day
+        queryExtraIndexTimeMatch = iname.match(/([0-9][0-9])(0[1-9]|1[0-2])(0[1-9]|[12][0-9]|3[01])$/);
+        if (queryExtraIndexTimeMatch) {
+          queryExtraIndexTimeMatched = true;
+          index = queryExtraIndexTimeMatch[0];
+        }
 
-      const start = Date.UTC(year, month - 1, day, hour) / 1000;
-      const stop = Date.UTC(year, month - 1, day, hour) / 1000 + len;
+        if (!queryExtraIndexTimeMatched){
+          // hourly 240311h19                     v year      v month        v day                    h  v hour
+          queryExtraIndexTimeMatch = iname.match(/([0-9][0-9])(0[1-9]|1[0-2])(0[1-9]|[12][0-9]|3[01])[Hh]([01][0-9]|2[0-3])$/);
+          if (queryExtraIndexTimeMatch) {
+            queryExtraIndexTimeMatched = true;
+            index = queryExtraIndexTimeMatch[0];
+          }
+        }
 
-      switch (bounding) {
-      default:
-      case 'last':
-        if (stop >= startTime && start <= stopTime) {
-          indices.push(iname);
+        if (!queryExtraIndexTimeMatched){
+            // weekly 24w10                       v year     w  v week
+          queryExtraIndexTimeMatch = iname.match(/([0-9][0-9])[Ww]([0-4][0-9]|5[0-3])$/);
+          if (queryExtraIndexTimeMatch) {
+            queryExtraIndexTimeMatched = true;
+            index = queryExtraIndexTimeMatch[0];
+          }
         }
-        break;
-      case 'first':
-      case 'both':
-      case 'either':
-      case 'database':
-        if (stop >= (startTime - len) && start <= (stopTime + len)) {
-          indices.push(iname);
+
+        if (!queryExtraIndexTimeMatched){
+            // monthly 24m10                      v year     w  v month
+          queryExtraIndexTimeMatch = iname.match(/([0-9][0-9])[Mm](0[1-9]|1[0-2])$/);
+          if (queryExtraIndexTimeMatch) {
+            queryExtraIndexTimeMatched = true;
+            index = queryExtraIndexTimeMatch[0];
+          }
         }
-        break;
+
+        // TODO: hourly2, ..., hourly12
+      } // if (isQueryExtraIndex)
+
+      if (!isQueryExtraIndex || queryExtraIndexTimeMatched) {
```

Comment:
`!isQueryExtraIndex` means all the regular default Arkime indices, or `queryExtraIndexTimeMatched` meaning it's a user-defined one but we were able to successfully glean the date/time out of it.